src/utils/getRegexp.ts

```diff
@@ -0,0 +1,101 @@
+import {FileEntry} from '@models/fileentry';
+
+export type MatchParamProps = {
+  matchCase: boolean;
+  matchWholeWord: boolean;
+  regExp: boolean;
+};
+
+export function getRegexp(query: string, params: MatchParamProps): RegExp {
```

Comment:
could you add a comment for this function with an example of input & output? something to show why this logic is needed